postgres_copy/psycopg_compat.py

```diff
@@ -0,0 +1,43 @@
+try:
+    import psycopg  # noqa: F401  just detect the presence of psycopg(3)
+    from codecs import getincrementaldecoder
+    from io import TextIOBase
+
+    BUFFER_SIZE = 128 * 1024
+
+    class NoopDecoder:
+        def decode(self, input, final=False):
+            return input
+
+    utf8_decoder_cls = getincrementaldecoder("utf8")
+
+    def copy_to(cursor, sql, params, destination):
+        # psycopg2.copy_expert has special handling of encoding
+        if isinstance(destination, TextIOBase):
+            decoder = utf8_decoder_cls()
+        else:
+            decoder = NoopDecoder()
+        with cursor.copy(sql, params) as copy:
+            while data := copy.read():
+                data = decoder.decode(data)
+                destination.write(data)
+            # TODO: is this extra one needed?
```

Comment:
this is needed. was going to fixup and squash but maybe best to make more commits and fix history after the review?

Reply:
I'm not sure I follow. If you want to make another patch yet before merge, let's get 'er in.

Reply:
I'd like to give them at least a beat, yeah. if we don't hear today we can revisit tomorrow.